core/net6.0/Dockerfile

```diff
@@ -26,9 +26,9 @@ RUN dotnet restore
 COPY proxy/Apache.OpenWhisk.Runtime.Common/. ./Apache.OpenWhisk.Runtime.Common/
 COPY proxy/Apache.OpenWhisk.Runtime.Dotnet.Minimal/. ./Apache.OpenWhisk.Runtime.Dotnet.Minimal/
 WORKDIR /app/Apache.OpenWhisk.Runtime.Dotnet.Minimal
-RUN dotnet publish -c Release -r linux-musl-x64 -o out
+RUN dotnet publish -c Release -o out
 
-FROM mcr.microsoft.com/dotnet/runtime:6.0-alpine AS runtime
+FROM mcr.microsoft.com/dotnet/aspnet:6.0-alpine AS runtime
```

Comment:
Can you confirm if changing this is necessary? Going from runtime to aspnet does increase the image size by ~ 20%.

Reply:
For transparency ...

mcr.microsoft.com/dotnet/aspnet: 117 MB
mcr.microsoft.com/dotnet/runtime: 93.8 MB

Reply:
Apologies, I needed to be more clear about the images that I was referring to.

I am comparing the current [openwhisk/action-dotnet-v6.0:1.18.0](https://hub.docker.com/layers/openwhisk/action-dotnet-v6.0/1.18.0/images/sha256-7b6275f129d2aeea3856bb7b769ff3e2f24220b23affa52d473b8f8f33967a6b?context=explore) (76.12MB) and the one [from this change](https://hub.docker.com/layers/gleonoliva/action-dotnet-v6.0/slim/images/sha256-9de9c2a172c4bfa0935b993bd997b063105f1d70fdf379d71f79b13104656abc?context=explore) (42.94 MB). I know that these are different architectures but I expect similar results for amd64.

This is because the step that copies the output from `dotnet publish -r` adds a bunch of libraries that are already in the `aspnet` image, under the `/usr/share/dotnet/shared/` directory to be specific. 

Here is a screenshot of the files bundled in the current image (1.18.0) under the `/app` directory.

![image](https://github.com/user-attachments/assets/0354da3f-9bbb-440b-ad51-4e783ac13c87)

Once I saw the dlls with "AspNetCore" in the name, I figured that maybe it would be better to switch to aspnet as the runtime image.

Reply:
I understood the app size differences, I'm not sold on switching the image yet. Lemme run a comparison on my side, hopefully today, to see if the image needs to be aspnet or if we can stick with runtime.

Reply:
Sure thing, I am not sure if this is the way to go for the project since I'm just getting involved with it, it just made sense to me. Hopefully we can find an approach that enables building on M1.
Thanks for taking the time to review.

Reply:
So I'm having no luck running the complete test suite locally (podman) and I'm not having any luck on my arm64 (Turing RK1) device. So I cannot test this out locally. Would you be able to change the image on your PR and see if the automated build tests pass?

Reply:
Do you mean changing it back to `runtime`? I can do that.

Reply:
I pushed the change but I cannot run the pipeline.

Reply:
I approved running it.